plugins/modules/pids.py

```diff
@@ -195,7 +195,7 @@ def __init__(self, module):
         self._pids = []
 
     def execute(self):
-        if self._name:
+        if self._name is not None:
```

Comment:
I think we should add a note to the docs of both `name` and `pattern` describing what happens when the value (for each one of them) is `""`. I _believe_ that for `name` it will bring all the processes of the target system and for `pattern` that's already the case. Which, IMHO, defeats the purpose of this module even though I can imagine a few good use cases, and many bad ones, for that "feature". Mixed feelings.